### PR TITLE
feat: add generated-files-bot

### LINKED ISSUE #1034: New Bot: autosynth managed files bot
synthtool manages a synth.metadata file which optionally tracks generated files. People often try to manually update a file managed by synthtool templates.

We can have a bot that warns a PR if you are modifying a file that is managed by synthtool templates.

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #1034

### CODE REVIEW

File: packages/template-bot/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+npm install
+
+# Run the bot
+npm start
+```
+
+### Configuration
+
+To configure the bot, you can create a configuration file:
+`.github/release-please.yml`. The contents of this file allow for the following
```
Comment (sofisl) [inline packages/template-bot/README.md:20]: should be template-bot.yml

**Comment by sofisl** [comment]
@chingor13, is there any way to cross-reference with synthtool templates as well, instead of having to provide a config file or external manifest?

**Comment by chingor13** [comment]
> Is there any way to cross-reference with synthtool templates as well, instead of having to provide a config file or external manifest?

For synthtool templates, we would use:
```yaml
externalManifests:
- type: json
  file: 'synth.metadata'
  jsonpath: '$.generatedFiles[*]'
```

Used external manifests so as to not hard-code to synthtool.

**Review by tbpg — COMMENTED** [review]
Naming comment: what about naming this bot something like "Generated Files" rather than "Template"? Files can be generated in a number of ways.

File: packages/template-bot/src/template-bot.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    const owner = context.payload.repository.owner.login;
+    const repo = context.payload.repository.name;
+    const pullNumber = context.payload.pull_request.number;
+
+    // Read the list of templated files
+    const templatedFiles = new Set(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      await getFileList(config, context.github as any, owner, repo)
```
[the remark this answers is not in the document]
Comment (chingor13) [reply packages/template-bot/src/template-bot.ts]: Looks like that works!

**Review by bcoe — APPROVED** [review]

**Pull request merged**